### PR TITLE
Copies of books

### PR DESCRIPTION
**Pull request by MaxTheComputerer**
- Books can be checked in and out
- Overdue books are highlighted and the numbers of days late is shown
- Still to do: Populate "Borrower" dropdown with list of library members (need changes from #4)

Will conflict with #4 because this still uses just the Home controller.

### CODE REVIEW

**Review by cec200wire — APPROVED** [review]

File: Bookish/Controllers/HomeController.cs [hunk 1]
```diff
@@ -78,17 +78,50 @@ public ActionResult AddBook(BookModel newBook)
...
-        var copies = BookCopyService.GetCopies(book);
-        var result = new BookCopyResult()
+        if (book == null || book.Id == null)
         {
-            book = book,
-            copies = copies
-        };
+            return RedirectToAction(nameof(Catalogue));
+        }
+        var result = BookCopyService.GetCopies(book);
```
Comment (emilypatsko) [inline Bookish/Controllers/HomeController.cs]: I like that this logic lives in a service! However the service itself should probably not live in the Models folder, but in a separate Services folder. I left some similar comments on Claire's PR about some of the classes which contained useful methods. I suggested renaming the classes to something like `EditBooksHelper` and putting them in a Helpers folder, but you two should probably chat and come up with a common solution.

File: Bookish/Models/BookCopyModel.cs [hunk 2]
```diff
@@ -5,21 +5,28 @@ public class BookCopyModel
...
     }
 }
 
 public class BookCopyResult
 {
     public BookModel book { get; set; }
     public List<BookCopyModel> copies { get; set; }
+}
+
+public class CheckOutCopyModel
```
Comment (emilypatsko) [inline Bookish/Models/BookCopyModel.cs]: I reckon this could go in a separate file

File: Bookish/Views/Shared/_BookTable.cshtml [hunk 3]
```diff
@@ -20,9 +20,8 @@
                 <td>@book.Author</td>
                 <td>@book.Title</td>
                 <td>
-                    @using (Html.BeginForm("ViewCopies", "Home", FormMethod.Post))
+                    @using (Html.BeginForm("ViewCopies", "Home", FormMethod.Get))
```
Comment (emilypatsko) [inline Bookish/Views/Shared/_BookTable.cshtml]: If you just want a button to direct to a specific controller action, you don't need a whole form - you can supply a button input with a `Url.Action`. Take a look at the top answer [here](https://stackoverflow.com/questions/2503923/html-button-calling-an-mvc-controller-and-action-method)

File: Bookish/Views/Shared/_CopyTable.cshtml [hunk 4]
```diff
@@ -7,25 +7,102 @@
...
+            <th scope="col">Overdue by</th>
+            <th></th>
+            <th></th>
         </tr>
         </thead>
         <tbody>
         @foreach (var copy in Model)
         {
-            <tr>
+            <tr class="@(!copy.IsAvailable() && copy.OverdueTime().Days >= 1 ? "table-danger" : "")">
```
Comment (emilypatsko) [inline Bookish/Views/Shared/_CopyTable.cshtml:20]: Nice touch!

**Review by emilypatsko — CHANGES_REQUESTED** [review]
Few small changes, but looks good so far!

**Comment by MaxTheComputerer** [comment]
> Few small changes, but looks good so far!

Changes have been made!

**Comment by MaxTheComputerer** [comment]
Resolved merge conflicts

File: Bookish/Views/Shared/_CopyTable.cshtml [hunk 5]
```diff
@@ -7,25 +7,102 @@
...
                     <td>@copy.Borrower.Name</td>
                     <td>@copy.DueDate</td>
+                    @if (copy.OverdueTime().Days >= 1)
+                    {
+                        <td>@copy.OverdueTime().Days days</td>
+                    }
+                    else
+                    {
+                        <td></td>
+                    }
```
Comment (emilypatsko) [inline Bookish/Views/Shared/_CopyTable.cshtml]: You could probably reduce this to one line using a ternary and string interpolation, something like:
```
<td>@(copy.OverdueTime().Days >= 1 ? $"{copy.OverdueTime().Days} days" : "")</td>
```

File: Bookish/Views/Shared/_CopyTable.cshtml [hunk 6]
```diff
@@ -7,25 +7,102 @@
...
+                </td>
             </tr>
         }
         </tbody>
     </table>
-</div>
\ No newline at end of file
+</div>
+
+<!-- Modal -->
```
Comment (emilypatsko) [inline Bookish/Views/Shared/_CopyTable.cshtml]: Nice use of commenting to make it clear what this is - I wonder if this modal block could be separated out into a partial view?

**Review by emilypatsko — APPROVED** [review]
I've left a couple comments - you don't _have_ to make these changes, but they might be nice to look into. Otherwise I've approved 😄

**Pull request merged**